Library/ViewModels/ProjectPamphletContentViewModelTests.swift

```diff
@@ -698,5 +702,225 @@ final class ProjectPamphletContentViewModelTests: TestCase {
+      .init(question: .whoAreYou, response: "response-3")
+    ])
+
+    let mockService = MockService(fetchProjectSummaryResult: .success(envelope))
+
+    let optimizelyClient = MockOptimizelyClient()
+      |> \.experiments .~ [
+        OptimizelyExperiment.Key.nativeMeProjectSummary.rawValue: OptimizelyExperiment.Variant.control.rawValue
+      ]
+
+    withEnvironment(apiService: mockService, optimizelyClient: optimizelyClient) {
+      self.vm.inputs.configureWith(value: (project, .discovery))
+      self.vm.inputs.viewDidLoad()
+
+      self.loadProjectIntoDataSourceProjectSummaryItems.assertValues([], "Nothing emits immediately.")
+      self.loadProjectIntoDataSourceProject.assertValues([], "Nothing emits immediately.")
+      self.loadProjectIntoDataSourceRefTag.assertValues([], "Nothing emits immediately.")
+      self.loadMinimalProjectIntoDataSource.assertValues([], "Nothing emits immediately.")
+
+      // Begin presentation. When presenting the project `animated` will be false since it is embedded in the
+      // navigator controller.
+      self.vm.inputs.viewWillAppear(animated: false)
+      self.vm.inputs.viewDidAppear(animated: false)
+
+      self.loadProjectIntoDataSourceProjectSummaryItems.assertValues([[]], "Emits [] as view appears")
+      self.loadProjectIntoDataSourceProject
+        .assertValues([project], "Load the full project into the data source.")
+      self.loadProjectIntoDataSourceRefTag.assertValues(
+        [.discovery], "Load the refTag into the data source."
+      )
+      self.loadMinimalProjectIntoDataSource.assertValues(
+        [], "Do not load the minimal version of the project."
+      )
+
+      // End presentation.
+      self.vm.inputs.viewWillAppear(animated: false)
+      self.vm.inputs.viewDidAppear(animated: false)
+
+      self.loadProjectIntoDataSourceProjectSummaryItems.assertValues([[]])
+      self.loadProjectIntoDataSourceProject
+        .assertValues([project], "Nothing new emits when the view is done.")
+      self.loadProjectIntoDataSourceRefTag.assertValues(
+        [.discovery], "Nothing new emits when the view is done."
+      )
+      self.loadMinimalProjectIntoDataSource.assertValues([], "Nothing new emits when the view is done.")
+
+      // Simulate a new version of the project coming through
+      self.vm.inputs.configureWith(value: (project, .discovery))
+
+      self.loadProjectIntoDataSourceProjectSummaryItems.assertValues([[], []])
+      self.loadProjectIntoDataSourceProject.assertValues(
+        [project, project], "The new project is loaded into data source"
+      )
+      self.loadProjectIntoDataSourceRefTag.assertValues(
+        [.discovery, .discovery], "The new refTag is loaded into data source"
+      )
+      self.loadMinimalProjectIntoDataSource.assertValues([], "Nothing new emits when the view is done.")
+
+      self.scheduler.advance()
+
+      self.loadProjectIntoDataSourceProjectSummaryItems.assertValues(
+        [[], [], []], "[] is returned"
```

Comment:
I know we've cursed using `skipRepeats` in the past, but I wonder if it could be a performance improvement to use it here -- any values triggered by the `projectSummaryValues` signal will cause `loadProjectPamphletContentDataIntoDataSource` to fire (since we're using `combineLatest`). Every time that signal fires, the whole project table view will reload.

Reply:
Ok I've added a filter to prevent any empty arrays being emitted from `projectSummaryRequestValues` after the prefixed empty array to prevent unnecessary reloading.